src/impl/load.c

```diff
@@ -0,0 +1,289 @@
+#include "load.h"
+
+#include "../ent/game_history.h"
+#include "../util/io_util.h"
+#include "../util/string_util.h"
+#include "config.h"
+#include "gcg.h"
+#include <ctype.h>
+#include <stdbool.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <regex.h>
+
+char *get_xt_gcg_string(const char *identifier, ErrorStack *error_stack) {
+  char *game_id_str = NULL;
+
+  // Check if this is a Cross-tables URL
+  if (strstr(identifier, "cross-tables.com") != NULL) {
+      // Extract game ID from URL parameter u=XXXXX using regex
+      regex_t regex;
+      regmatch_t matches[2];
+      
+      int regex_status = regcomp(&regex, "u=([0-9]+)", REG_EXTENDED);
+      if (regex_status != 0) {
+          error_stack_push(error_stack, ERROR_STATUS_CONFIG_LOAD_MISSING_ARG,
+                          string_duplicate("Failed to compile regex for cross-tables URL parsing"));
+          return NULL;
+      }
+      
+      int match_result = regexec(&regex, identifier, 2, matches, 0);
+      if (match_result == 0) {
+          int start = matches[1].rm_so;
+          int end = matches[1].rm_eo;
+          int id_len = end - start;
+          
+          game_id_str = malloc_or_die(id_len + 1);
+          strncpy(game_id_str, identifier + start, id_len);
+          game_id_str[id_len] = '\0';
+      } else {
+          regfree(&regex);
+          error_stack_push(error_stack, ERROR_STATUS_CONFIG_LOAD_MISSING_ARG,
+                          get_formatted_string("Failed to extract game ID from cross-tables URL: %s", identifier));
+          return NULL;
+      }
+      
+      regfree(&regex);
+  } else {
+      // Check if it's a numeric game id
+      if (!is_all_digits_or_empty(identifier)){
+        return NULL;
+      }
+      game_id_str = string_duplicate(identifier);
+  }
+
+  char *gcg_content = NULL;
+
+  if (game_id_str) {
+      // Get first 3 digits for the path
+      char first_three[4] = {0};
+      for (int i = 0; i < 3 && game_id_str[i]; i++) {
+          first_three[i] = game_id_str[i];
+      }
+      
+      // Build the curl command directly
+      char *curl_cmd = get_formatted_string("curl -s -L \"https://cross-tables.com/annotated/selfgcg/%s/anno%s.gcg\"", 
+                                           first_three, game_id_str);
+      
+      gcg_content = get_process_output(curl_cmd);
+      
+      free(curl_cmd);
+      free(game_id_str);
+  }
+
+  if (!gcg_content || is_string_empty_or_whitespace(gcg_content) || 
+      strstr(gcg_content, "#player1") == NULL) {
+      free(gcg_content);
+      // Only push error if this was actually a cross-tables URL
+      if (strstr(identifier, "cross-tables.com") != NULL) {
```

Comment:
If the GCG content is screwed up we should probably return an error regardless of the identifier type. Some new error enum like "bad xt gcg" and a message saying the id, so that the error makes sense for both URL and game id input types.